### PR TITLE
Koa-5408 Added a search bar in the demo app

### PR DESCRIPTION
**Pull request by jaanainkinen**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt [hunk 1]
```diff
@@ -78,21 +89,45 @@ fun ComponentListScreen(
                 ),
             ),
         )
+        var textState by rememberSaveable { mutableStateOf("") }
```
Comment (bvitaliyg) [inline app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt:92]: searchQuery, perhaps?

File: app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt [hunk 2]
```diff
@@ -78,21 +89,45 @@ fun ComponentListScreen(
...
+            value = textState.replaceFirstChar { it.uppercase() },
+            onValueChange = { value ->
+                textState = value
+            },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(BpkSpacing.Base),
+            icon = BpkIcon.Search,
+        )
+        val filteredTokens = repository.tokenComponents.filter { it: Component ->
```
Comment (bvitaliyg) [inline app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt:104]: No need to specify it in these lambdas

**Review by bvitaliyg — COMMENTED** [review]
Couple of small comments

File: app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt [hunk 3]
```diff
@@ -78,21 +89,45 @@ fun ComponentListScreen(
                 ),
             ),
         )
+        var textState by rememberSaveable { mutableStateOf("") }
+        BpkTextField(
+            placeholder = stringResource(R.string.navigation_search),
+            value = textState.replaceFirstChar { it.uppercase() },
```
[the remark this answers is not in the document]
Comment (jaanainkinen) [reply app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt:95]: @marianeum advised that the search text should be case insensitive, and this was my attempt to do that. Without it, it wouldn't recognise for example "badge", so with this it automatically turns the first character into uppercase. Is there a better way to do it?

File: app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt (hunk 1 above)
Comment (jaanainkinen) [reply app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt:92]: Thanks, renamed it!

File: app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt (hunk 2 above)
Comment (jaanainkinen) [reply app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt:104]: Removed them.

File: app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaanainkinen) [reply app/src/main/java/net/skyscanner/backpack/demo/ui/ComponentListScreen.kt]: This is fixed with ignoreCase.

**Pull request merged**